File: Python/underautomation/universal_robots/primary_interface/version_event_args.py
import typing
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import VersionEventArgs as version_event_args

class VersionEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = version_event_args()
		else:
			self._instance = _internal
	@property
	def project_name(self) -> str:
		return self._instance.ProjectName
	@project_name.setter
	def project_name(self, value: str):
		self._instance.ProjectName = value
	@property
	def major_version(self) -> int:
		return self._instance.MajorVersion
	@major_version.setter
	def major_version(self, value: int):
		self._instance.MajorVersion = value
	@property
	def minor_version(self) -> int:
		return self._instance.MinorVersion
	@minor_version.setter
	def minor_version(self, value: int):
		self._instance.MinorVersion = value
	@property
	def bugfix_version(self) -> int:
		return self._instance.BugfixVersion
	@bugfix_version.setter
	def bugfix_version(self, value: int):
		self._instance.BugfixVersion = value
	@property
	def build_number(self) -> int:
		return self._instance.BuildNumber
	@build_number.setter
	def build_number(self, value: int):
		self._instance.BuildNumber = value
	@property
	def build_date(self) -> str:
		return self._instance.BuildDate
	@build_date.setter
	def build_date(self, value: str):
		self._instance.BuildDate = value
	@property
	def version(self) -> typing.Any:
		return self._instance.Version

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/VersionEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Tool data
	/// </summary>
	public class VersionEventArgs : PackageEventArgs {

		/// <summary>
		/// URControl project
		/// </summary>
		[PackageDescription("URControl project")]
public string ProjectName { get; set; }

		/// <summary>
		/// Major version number, for example 5 in 5.6.1.1234
		/// </summary>
		[PackageDescription("Major version number")]
public byte MajorVersion { get; set; }

		/// <summary>
		/// Minor firmware version number, for example 6 in 5.6.1.1234
		/// </summary>
		[PackageDescription("Minor firmware version number")]
public byte MinorVersion { get; set; }

		/// <summary>
		/// Firmware bugfix number, for example 1 in 5.6.1.1234
		/// </summary>
		[PackageDescription("Firmware bugfix number")]
public int BugfixVersion { get; set; }

		/// <summary>
		/// Firmware build number, for example 1234 in 5.6.1.1234
		/// </summary>
		[PackageDescription("Firmware build number")]
public int BuildNumber { get; set; }

		/// <summary>
		/// Build date of the firmware, for example "DEC 2020"
		/// </summary>
		[PackageDescription("Build date of the firmware")]
public string BuildDate { get; set; }

		/// <summary>
		/// Version of the firmware
		/// </summary>
		public Version Version { get; }
	}
}
